eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version4/ChannelTypes4.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.wire.internal.channel.version4
+
+import fr.acinq.eclair.channel.LocalFundingStatus.ConfirmedFundingTx
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.wire.protocol.{ChannelAnnouncement, ChannelUpdate, Shutdown}
+import fr.acinq.eclair.{Alias, RealShortChannelId}
+
+private[channel] object ChannelTypes4 {
+
+  // We moved the real scid inside each commitment object when adding DATA_NORMAL_14_Codec.
+  case class ShortIds(real_opt: Option[RealShortChannelId], localAlias: Alias, remoteAlias_opt: Option[Alias])
+
+  // We moved the channel_announcement inside each commitment object when adding DATA_NORMAL_14_Codec.
+  case class DATA_NORMAL_0e(commitments: Commitments,
+                            shortIds: ShortIds,
+                            channelAnnouncement: Option[ChannelAnnouncement],
+                            channelUpdate: ChannelUpdate,
+                            localShutdown: Option[Shutdown],
+                            remoteShutdown: Option[Shutdown],
+                            closingFeerates: Option[ClosingFeerates],
+                            spliceStatus: SpliceStatus) {
+    def migrate(): DATA_NORMAL = {
+      val commitments1 = commitments.copy(
+        active = commitments.active.map(c => setScidIfMatches(c, shortIds)),
+        inactive = commitments.inactive.map(c => setScidIfMatches(c, shortIds)),
+      )
```

Comment:
Just for fun, I know that you are not a fan of quicklens, this would be:

```scala
commitments
        .modifyAll(
          _.active.each.localFundingStatus.when[ConfirmedFundingTx].shortChannelId,
          _.inactive.each.localFundingStatus.when[ConfirmedFundingTx].shortChannelId
      ).setToIfDefined(shortIds.real_opt)
```